<a href="https://colab.research.google.com/github/akbarafriansyah/data-science-2026/blob/main/Pertemuan2_Afriansyah_Akbar_250401020013.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [27]:
#Praktikum Pertemuan 2
import pandas as pd
import numpy as np

#Muat Dataset Titanic
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)

#Eksplorasi Awal
print(df.shape, df.columns.tolist())
print(df.isnull().sum())

# 3 Filtering — Penumpang Wanita Kelas 1
wk1 = df[(df['Sex']=='female') &
(df['Pclass']==1)]

print(wk1['Survived'].mean(), wk1['Age'].mean())

# 4 Groupby — Tingkat Keselamatan per Kelas
hasil = df.groupby('Pclass').agg(
total=('PassengerId','count'),
selamat=('Survived','mean'),
).round(3)
print(hasil)


(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64
0.9680851063829787 34.61176470588235
        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


In [37]:
#Q1
print("Jumlah Baris dan Kolom     : ", df.shape)
print("Kolom yang tersedia adalah : ", df.columns.tolist())

Jumlah Baris dan Kolom     :  (891, 12)
Kolom yang tersedia adalah :  ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


Dataset Titanic memiliki **891 baris dan 12 kolom**.
Artinya terdapat 891 data penumpang dengan 12 variabel yang mendeskripsikan setiap penumpang.

In [57]:
#Q2
missing = df.isnull().sum()
percent = (missing / len(df)) * 100

pd.DataFrame({
    'Jumlah Missing': missing,
    'Persentase (%)': percent
})

,Jumlah Missing,Persentase (%)
PassengerId,0,0.000000
Survived,0,0.000000
Pclass,0,0.000000
Name,0,0.000000
Sex,0,0.000000
Age,177,19.865320
SibSp,0,0.000000
Parch,0,0.000000
Ticket,0,0.000000
Fare,0,0.000000


Ditemukan bahwa kolom **Age, Cabin, dan Embarked** memiliki nilai kosong.
Kolom Cabin memiliki jumlah missing paling banyak.

1.   Cabin (77.10%)
2.   Age (19.86%)
3.   Embarkked (0.22%)



In [46]:
#Q3
print("Persentase penumpang yang selamat sebanyak", df['Survived'].mean() * 100,"%")

Persentase penumpang yang selamat sebanyak 38.38383838383838 %


Berdasarkan hal tersebut didapat sekitar **38,38%** penumpang berhasil selamat dari kecelakaan Titanic.

In [63]:
#Q4
female_class1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]

jumlah = female_class1.shape[0]
rata_usia = female_class1['Age'].mean()

print("Jumlah Penumpang Wanita dari kelas 1 berjumlah",jumlah,"orang, dan rata-rata usia yang selamat adalah sebesar", rata_usia,"%")

Jumlah Penumpang Wanita dari kelas 1 berjumlah 94 orang, dan rata-rata usia yang selamat adalah sebesar 34.61176470588235 %


Hasil yang didapat adalah Jumlah Penumpang wanita kelas 1 yang selamat adalah **94 orang** dan rata - rata usia adalah sekitar 34.9 Tahun atau Rata-rata usia apat dibulatkan menjadi **35 tahun.**

In [75]:
#Q5
survival_by_class = df.groupby('Pclass')['Survived'].mean()*100
print("Kelas Penumpang dengan tingkat keselamat tertinggi adalah kelas (dalam %):")
print("")
survival_by_class


Kelas Penumpang dengan tingkat keselamat tertinggi adalah kelas (dalam %):



,Survived
Pclass,
1,62.962963
2,47.282609
3,24.236253


Kelas 1 memiliki tingkat keselamatan tertinggi dibandingkan kelas lainnya yaitu dengan nilai rata - rata **0.63%**
didapat juga kesimpulan bahwa semakin tinggi kelas, semakin besar kemungkinan untuk selamat.

In [76]:
#Q6
print("Variabel yang saya ambil merupakan tingkat keselamatan dalam kategori jenis kelamin:")
print("")
survival_by_gender = df.groupby('Sex')['Survived'].mean()*100
survival_by_gender

Variabel yang saya ambil merupakan tingkat keselamatan dalam kategori jenis kelamin:



,Survived
Sex,
female,74.203822
male,18.890815


Interpretasi saya berdasarkan analisis ini adalah:
Wanita jauh lebih banyak selamat dibanding pria,
Ini menunjukkan adanya kebijakan "women and children first" saat evakuasi
Faktor sosial sangat berpengaruh, bukan hanya kelas ekonomi.